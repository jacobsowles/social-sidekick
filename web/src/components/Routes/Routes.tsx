import PropTypes from 'prop-types';
import React, { Component, ComponentState } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';

import Auth0CallbackPage from '@pages/Auth0CallbackPage';
import Auth0PostAuthPage from '@pages/Auth0PostAuthPage';
import ConnectionsPageContainer from '@pages/ConnectionsPageContainer';
import ContactPage from '@pages/ContactPage';
import HomePage from '@pages/HomePage';
import LandingPage from '@pages/LandingPage';
import OAuthCallbackPage from '@pages/OAuthCallbackPage';
import AuthService from '@core/auth';
import PrivateRoute from './PrivateRoute';
import './Routes.scss';

class Routes extends Component {
  public static contextTypes = {
    router: PropTypes.object
  };

  private authService = new AuthService();

  public handleAuthentication = (nextState: ComponentState) => {
    if (/access_token|id_token|error/.test(nextState.location.hash)) {
      this.authService.handleAuthentication((location: string) =>
        this.context.router.history.replace(location)
      );
    }
  };

  public render() {
    return (
      <div className="container">
        <Switch>
          <Route
            exact
            path="/"
            render={() =>
              this.authService.isAuthenticated() ? <Redirect to="/home" /> : <LandingPage />
            }
          />
          <Route
            path="/auth0-callback"
            render={props => (
              <Auth0CallbackPage handleAuthentication={this.handleAuthentication} {...props} />
            )}
          />
          <PrivateRoute path="/connections" component={ConnectionsPageContainer} />
          <Route path="/contact" component={ContactPage} />
          <PrivateRoute path="/home" component={HomePage} />
          <PrivateRoute path="/oauth-callback" component={OAuthCallbackPage} />} />
          <Route path="/auth0-postauth" component={Auth0PostAuthPage} />
        </Switch>
      </div>
    );
  }
}

export default Routes;
