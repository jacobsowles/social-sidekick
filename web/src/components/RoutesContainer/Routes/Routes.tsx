import React, { ComponentState, FunctionComponent } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';

import Auth0CallbackPage from '@pages/Auth0CallbackPage';
import Auth0PostAuthPage from '@pages/Auth0PostAuthPage';
import ConnectionsPage from '@pages/ConnectionsPage';
import ContactPage from '@pages/ContactPage';
import DashboardPage from '@pages/DashboardPage';
import LandingPage from '@pages/LandingPage';
import OAuthCallbackPage from '@pages/OAuthCallbackPage';
import AuthService from '@core/auth';
import { UserService } from '@core/types';
import PrivateRoute from './PrivateRoute';
import './Routes.scss';

export interface RoutesProps {
  handleAuthentication: (nextState: ComponentState) => void;
  serviceModules: any[] | null;
  services: UserService[] | null;
  userId?: string;
}

const Routes: FunctionComponent<RoutesProps> = ({
  handleAuthentication,
  serviceModules,
  services,
  userId
}) => {
  const authService = new AuthService();

  return (
    <div className="container">
      <Switch>
        <Route
          exact
          path="/"
          render={() =>
            authService.isAuthenticated() ? <Redirect to="/dashboard" /> : <LandingPage />
          }
        />
        <Route
          path="/auth0-callback"
          render={props => (
            <Auth0CallbackPage handleAuthentication={handleAuthentication} {...props} />
          )}
        />
        <Route path="/auth0-postauth" component={Auth0PostAuthPage} />

        {authService.isAuthenticated && (
          <Route
            path="/connections"
            render={() => <ConnectionsPage services={services} userId={userId} />}
          />
        )}

        {!authService.isAuthenticated && (
          <Route
            path="/connections"
            render={() => {
              authService.login();
              return null;
            }}
          />
        )}

        <Route path="/contact" component={ContactPage} />

        {authService.isAuthenticated && (
          <Route
            path="/dashboard"
            render={() => (
              <DashboardPage serviceModules={serviceModules} services={services} userId={userId} />
            )}
          />
        )}

        {!authService.isAuthenticated && (
          <Route
            path="/dashboard"
            render={() => {
              authService.login();
              return null;
            }}
          />
        )}

        <PrivateRoute path="/oauth-callback" component={OAuthCallbackPage} />
      </Switch>
    </div>
  );
};

export default Routes;
