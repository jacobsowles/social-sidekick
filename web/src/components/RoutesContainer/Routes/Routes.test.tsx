import React from 'react';
import { Route, Switch } from 'react-router-dom';

import { init, shallow, toJson } from '@tests/component-test-base';
import Routes, { RoutesProps } from './Routes';

jest.mock(
  '@core/auth/auth.service',
  () =>
    class MockAuthService {
      public login = jest.fn();
      public isAuthenticated = () => true;
    }
);

describe('Routes', () => {
  let props: RoutesProps;

  init(() => <Routes {...props} />);

  beforeEach(() => {
    props = {
      handleAuthentication: jest.fn(),
      serviceModules: null,
      services: null
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  it('should always render a container', () => {
    shallow()
      .find('.container')
      .should.have.lengthOf(1);
  });

  it('should always render a Switch component', () => {
    shallow()
      .find(Switch)
      .should.have.lengthOf(1);
  });

  it('should always render a Route component', () => {
    shallow()
      .find(Route)
      .should.have.length.greaterThan(0);
  });
});
