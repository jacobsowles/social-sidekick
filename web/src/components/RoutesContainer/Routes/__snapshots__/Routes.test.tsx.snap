// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render the component without crashing 1`] = `
<div
  className="container"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/auth0-callback"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/auth0-postauth"
    />
    <Route
      path="/connections"
      render={[Function]}
    />
    <Route
      component={[Function]}
      path="/contact"
    />
    <Route
      path="/dashboard"
      render={[Function]}
    />
    <PrivateRoute
      component={[Function]}
      path="/oauth-callback"
    />
  </Switch>
</div>
`;
